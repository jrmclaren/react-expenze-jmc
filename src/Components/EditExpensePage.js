import React from 'react';
import { connect } from 'react-redux';
import { editExpense, startRemoveExpense } from '../Actions/expenses';
import ExpenseForm from './ExpenseForm';


// Refactor to class based component 
// setup mapDispatchToProps ( editExpense, startRemoveExpense ) => dispatch


export class EditExpensePage extends React.Component {
  constructor(props) {
    super(props);
  }
  startRemoveExpense =  e => {
    this.props.startRemoveExpense({
      id: this.props.expense.id
    });
    this.props.history.push('/');
  }
  editExpense = expense => {
    this.props.onSubmit(this.props.expense.id, expense);
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
				<ExpenseForm
      expense={this.props.expense}
      onSubmit={this.editExpense}
      />
			<button  onClick={this.startRemoveExpense}>Remove</button> 
		</div>
      );
  }
}
const mapDispatchToProps = (dispatch, props) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  startRemoveExpense: expense => dispatch(startRemoveExpense(expense))
})
const mapStateToProps = (state, props) => {
  return {
    expense: ( state.expenses.find(expense => expense.id === props.match.params.id) )
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);


