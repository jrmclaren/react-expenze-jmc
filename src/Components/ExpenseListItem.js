import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ( {description, amount, createdAt, id} = {} ) => (
			<div>
				<hr />
				<Link to={`/edit/${id}`}>
					<h3>{description}</h3>
				</Link>
				<p>Amount: {amount}</p>
				<p> Timestamp: {createdAt}</p>

			</div> 
		);
export default ExpenseListItem;
