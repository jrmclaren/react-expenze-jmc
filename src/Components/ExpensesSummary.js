import React from 'react';
import numeral from 'numeral';
import { selectExpensesTotal } from '../Selectors/expenses-total';
import selectExpenses from '../Selectors/expenses';
import { connect } from 'react-redux';

const ExpensesSummary = ({expenseCount, expensesTotal}) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
  		<h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
  	</div>
    );
}
const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
};
export { ExpensesSummary }
export default connect(mapStateToProps)(ExpensesSummary);