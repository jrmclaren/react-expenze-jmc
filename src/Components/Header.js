import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../Actions/auth';

export const Header = ({ startLogout }) => (
  <header>
			<h1> Expenszes </h1>
			<NavLink to='/' activeClassName={"is-active"} exact>Home</NavLink>
			<br />
			<NavLink to='/create' activeClassName={"is-active"} >New Expense</NavLink>
			<br />
			<button onClick={startLogout}>Logout</button> 
			<div>Â </div>
		</header>
);

const mapDispatchToProps = dispatch => ({
	startLogout: () => dispatch(startLogout())
});
export default  connect(undefined, mapDispatchToProps)(Header);

