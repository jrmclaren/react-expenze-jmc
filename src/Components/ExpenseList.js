import React from 'react';
import ExpenseListItem from './ExpenseListItem';
import { connect } from 'react-redux';
import selectExpenses from '../Selectors/expenses';

export const ExpenseList = props => (
  <div>	
  	{
  props.expenses.length === 0 ? (
    <p>No Expenses </p>	) : (
    props.expenses.map((expense, key) => {
      return ( <ExpenseListItem key={key} {...expense} /> );
    })
    )
  }		
 </div>
);

const mapStateToProps = state => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  }
};

export default connect(mapStateToProps)(ExpenseList);