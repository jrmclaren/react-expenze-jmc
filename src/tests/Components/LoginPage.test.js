import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../Components/LoginPage';

test('should render login page with button', () => {
  const wrapper = shallow(<LoginPage  />);
  expect(wrapper).toMatchSnapshot();
});

test(' should call startLogin on button Click ', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage  startLogin={startLogin} />);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});

