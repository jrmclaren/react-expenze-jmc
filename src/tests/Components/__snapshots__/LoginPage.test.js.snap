// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page with button 1`] = `
<div>
  <button
    type="button"
  >
    Login
  </button>
</div>
`;
